src/java_bytecode/java_bytecode_convert_class.cpp

```diff
@@ -246,6 +246,11 @@ void java_bytecode_convert_classt::convert(
 /// java::array[X] where X is byte, float, int, char...
 void java_bytecode_convert_classt::add_array_types()
 {
+  // this method does not need to be called more than once
+  static bool already_executed=false;
+  if(already_executed)
+    return;
+
```

Comment:
Thanks for the suggestions.

I made it a static method of the same class (I find that's the right place for it), added one static Boolean field to the class and used two calls to `INVARIANT` to make sure that the user code firsts calls the static method and then does the conversion bytecode -> codet *after* having called the static method.